lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -613,8 +617,9 @@ def get_facts(self):
             'update_order': self.update_order}
 
     @staticmethod
-    def from_ansible_params(ap, old_service):
+    def from_ansible_params(ap, old_service, image_digest):
         s = DockerService()
+        s.image = image_digest
         s.constraints = ap['constraints']
         s.placement_preferences = ap['placement_preferences']
         s.image = ap['image']
```

Comment:
Now you overwrite `s.image` you just set a few lines before.

Reply:
Oh wow, must have missed it in the rebase.